torchrl/collectors/collectors.py

```diff
@@ -585,7 +585,7 @@ def _reset_if_necessary(self) -> None:
             else:
                 self._tensordict.zero_()
 
-            self._tensordict.update(self.env.reset(), inplace=True)
+            self._tensordict.update(self.env.reset(self._tensordict), inplace=True)
```

Comment:
I think we don't need the call to update.
env.reset() will update the tensordict.